router/router.go

```diff
@@ -53,21 +55,17 @@ type Router struct {
 	acceptLimiter   *TokenBucket
 }
 
-type PacketSource interface {
-	ReadPacket() ([]byte, error)
-}
+func NewRouter(config Config, name PeerName, nickName string) *Router {
+	router := &Router{Config: config, gossipChannels: make(GossipChannels)}
 
-type PacketSink interface {
-	WritePacket([]byte) error
-}
+	if router.Bridge == nil {
+		router.Bridge = NullBridge{}
+	}
 
-type PacketSourceSink interface {
-	PacketSource
-	PacketSink
-}
+	if router.Overlay == nil {
+		router.Overlay = NullOverlay{}
+	}
```

Comment:
Are you sure about that?  I would agree that we shouldn't modify the caller-provided `Config` here, but that's not what we are doing.  We are modifying fields in `Router` that happen to be inherited from `Config`.  So how would a `NewConfig` help here?